TundraTD/Assets/Scripts/Mobs/MobsBehaviour/Squirrel/SquirrelBehaviour.cs

```diff
@@ -0,0 +1,111 @@
+using System.Collections;
+using Spells;
+using UnityEngine;
+
+namespace Mobs.MobsBehaviour.Squirrel
+{
+    [RequireComponent(typeof(MobModel))]
+    public class SquirrelBehaviour : MobBehaviour
+    {
+        private Vector3 _finalTreeCords;
+        private Vector3 _treeCords;
+        private bool _isFirstTree = true;
+        private bool _goToGates = true;
+        private bool _isTreeNotTouched = true;
+        [SerializeField] private float mobShield;
+
+
+        public override BasicElement MobBasicElement => BasicElement.Earth;
+        public override BasicElement MobCounterElement => BasicElement.Air;
+
+
+        protected override void HandleIncomeDamage(float damage, BasicElement damageElement)
+        {
+            var multiplier = 1f;
+            if (damageElement == MobBasicElement)
+                multiplier = 0.8f;
+            else if (damageElement == MobCounterElement)
+                multiplier = 1.2f;
+
+            MobModel.CurrentMobHealth -= damage * multiplier;
+        }
+
+        public override void ExecuteOnMobSpawn(Transform gates, MobPortal mobPortal)
+        {
+            MobPortal = mobPortal;
+            MobModel.InstantiateMobModel();
+
+            DefaultDestinationPoint = gates;
+            MobModel.MobNavMeshAgent.enabled = true;
+            StartCoroutine(FindTree());
+            
+        }
+
+        private IEnumerator FindTree()
+        {
+            float _minDistance = 10000;
+            Collider[] _allTrees = Physics.OverlapSphere(this.transform.position, 1000);
+            foreach(var _tree in _allTrees)
+            {
+                if (_isFirstTree)
+                {
+                    if (_tree.gameObject.CompareTag("Tree"))
+                    {
+                        float _distanceToTree = Vector3.Distance(this.transform.position, _tree.gameObject.transform.position);
+                        if (_distanceToTree < _minDistance)
+                        {
+                            _minDistance = _distanceToTree;
+                            _treeCords = _tree.gameObject.transform.position;
+                            _goToGates = false;
+                        }
+                    }
+                }
+                else
+                {
+                    if (_tree.gameObject.CompareTag("Tree") && _finalTreeCords != _tree.transform.position
+                        && this.transform.position.z < _tree.transform.position.z
```

Comment:
```suggestion
Vector3.Dot(transform.position  - _tree.transform.position, _tree.transform.position) > 0
```
> 0 спереди
< 0 сзади
= 0 сбоку